### PR TITLE
feat(plugins): add peertubeHelpers.loadWithFiles

### PR DESCRIPTION
**Pull request by kontrollanten**
## Description
Let plugin authors to load a video with all its files. This can be useful when a plugin want to switch video files based on the users capabilities.

## Has this been tested?

- [x] 👍 yes, I added tests to the test suite

### CODE REVIEW

File: server/core/types/plugins/register-server-option.model.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ export type PeerTubeHelpers = {
 
   videos: {
     loadByUrl: (url: string) => Promise<MVideoThumbnail>
+    loadWithFiles: (id: number | string) => Promise<MVideoWithAllFiles>
```
Comment (Chocobozzz) [inline server/core/types/plugins/register-server-option.model.ts:37]: `loadByIdOrUUIDWithFiles`

**Comment by Chocobozzz** [comment]
`getAllFiles` helper don't do the job for your use case?

**Comment by kontrollanten** [comment]
Yes, I need to call `getHLSPlaylist()` to get URL to master playlist and segments file.

**Comment by Chocobozzz** [comment]
> Yes, I need to call getHLSPlaylist() to get URL to master playlist and segments file.

Okay then, but `getHLSPlaylist` is not considered as a plugin API and can break at any moment.

Don't you prefer to create a dedicated plugin helper, like we did for video files, to fetch HLS playlist information?

**Comment by Chocobozzz** [comment]
> That would be nice, but I think this use case is too niche to deserve its own plugin API. We use it like this:

Thanks! `filter:api.video.get.result` filter hook cannot be used to change HLS playlist URLs?

**Comment by kontrollanten** [comment]
> > That would be nice, but I think this use case is too niche to deserve its own plugin API. We use it like this:
> 
> Thanks! `filter:api.video.get.result` filter hook cannot be used to change HLS playlist URLs?

It can, that's where we're running the code I pasted above. We return different playlists depending whether the user is a "premium user" or not.

**Pull request merged**